js/main.js

```diff
@@ -1,44 +1,121 @@
 $photoUrl.addEventListener('input', function (event) {
-  // check the state of the value of the photo. if the value is empty or an '', then the attribute should be the placeholder. otherwise, it should hold the value of the photo URL
   if ($photoUrl.value === '') {
     $imgSrc.setAttribute('src', './images/placeholder-image-square.jpg');
   } else {
     $imgSrc.setAttribute('src', $photoUrl.value);
   }
 });
 
-var $entryForm = document.querySelector('.entry-form'); // must assign '.entry-form' to a new variable first
-$entryForm.addEventListener('submit', newEntry); // listen for 'submit' events on the journal entry form. newEntry is made as a name for our function which will be stated in the next line.
-
-function newEntry(event) { // function is defined and named here called function newEvent(event) { }
+function newEntry(event) {
   event.preventDefault(); // event.preventDefault() will prevent the browser from automatically reloading the page with the form's values in the url.
-  var title = document.querySelector('.title').value; // select '.title' and assign it to a new variable.
-  var photoURL = document.querySelector('.photo').value; // select '.photo' and assign it to a new variable.
-  var notes = document.querySelector('.textbox').value; // select '.textbox' and assign it to a new variable.
-  var entryData = { // using JavaScript to gather user input from HTML forms using var entryData = { }.
-    // the title, photoURL, notes (the new variables) will be assigned to a new variable as a collective. This will be called entryData
-    title, // this is the variable we assigned to title.values and so on for the next lines 29 and 30.
-    photoURL,
-    notes
+  var oldEntries = localStorage.getItem('JavaScriptLocalStorage');
+  oldEntries = JSON.parse(oldEntries);
+  var $title = document.querySelector('.title').value;
+  var $photoURL = document.querySelector('.photo').value;
+  var $notes = document.querySelector('.textbox').value;
+  var entryData = {
+    title: $title,
+    photoURL: $photoURL,
+    notes: $notes
   };
-  entryData.entryId = data.nextEntryId;
+  entryData.entryId = data.nextEntryId; // creating a new key for entryData
+  entryList.prepend(userEntry(entryData));
   data.nextEntryId++; // incrementing
-  data.entries.unshift(entryData); // prepends. prepends means to put it in the beginning of something which is used as a .unshift();. With each data SAVED, it will be placed in the beginning of the object.
-  // localStorage.setItem('New Data', JSON.stringify(entryData)); // storing in localstorage by using localStorage.setItem(). Must use JSON.stringify() because localStorage only accepts data as a string.
+  data.entries.unshift(entryData);
+  localStorage.setItem('JavaScriptLocalStorage', JSON.stringify({ ...data, entries: [entryData, ...oldEntries.entries] })); // entries: [entryData, ...oldEntries.entries] == reassigning entries to an array with all of the entries of the oldEntries and adding onto entryData
```

Comment:
You should only be updating localStorage right before the page unloads, not with every entry submission.